synapse/handlers/federation.py

```diff
@@ -796,9 +796,15 @@ async def _handle_marker_event(self, origin: str, marker_event: EventBase):
             return
 
         # Skip processing a marker event if the room version doesn't
-        # support it.
+        # support it or the event is not from the room creator.
         room_version = await self.store.get_room_version(marker_event.room_id)
-        if not room_version.msc2716_historical:
+        create_event = await self.store.get_create_event_for_room(marker_event.room_id)
+        room_creator = create_event.content.get("creator", None)
```

Comment:
```suggestion
        room_creator = create_event.content.get("creator")
```

is equivalent fwiw

Reply:
👍 We can also use the all new `EventContentFields.ROOM_CREATOR` from https://github.com/matrix-org/synapse/pull/10697 too